docs/backend/schema.ddl

```diff
@@ -0,0 +1,115 @@
+///////////////////////////////////////////////////
+// DPHs and users
+
+Table Dph {
+  id int [pk, increment]
+  name varchar [not null, unique]
+  gisaid_division_name varchar [unique]
+}
+
+Table User {
+  id int [pk, increment]
+  name varchar [not null]
+  auth0_user_id varchar [not null]
+  dph int [not null, ref: > Dph.id]
+  dph_admin bool [not null]
+}
+
+// members of dph can see data uploaded by owner_dph
+Table DphCanSee {
+  dph int [not null, unique, ref: > Dph.id]
+  owner_dph int [not null, unique, ref: > Dph.id]
+}
+
+///////////////////////////////////////////////////
+// Physical Samples
+
+Table PhysicalSample {
+  id int [pk, increment]
+  submitting_dph int [not null, ref: > Dph.id]
+  date_taken datetime [not null]
+  external_accession varchar [not null]
+  location varchar [not null]
+  extraction_method varchar [not null]
+  sample_source varchar [not null]
+  
+  // TODO: the other fields from PHA4GE
+  
+  Indexes {
+    (submitting_dph, external_accession) [unique]
+  }
+}
+
+Table Gene {
+  id int [pk, increment]
+  name varchar [not null, unique]
+}
+
+Table PcrPrimer {
+  id int [pk, increment]
+  gene_id int [not null, ref: > Gene.id]
+  name varchar [not null]
+  
+  Indexes {
+    (gene_id, name) [unique]
+  }
+}
+
+Table PcrCtValue {
+  physical_sample_id int [not null, ref: > PhysicalSample.id]
+  pcr_primer_id int [not null, ref: > PcrPrimer.id]
+  value float [not null]
+  
+  Indexes {
+    (physical_sample_id, pcr_primer_id) [unique]
+  }
+}
+
+///////////////////////////////////////////////////
+// Sequences
+
+// describes a single set of sequence data from the sequencer. (FASTQ)
+Table RawSequence {
+  id int [pk, increment]
+  physical_sample_id int [not null, ref: > PhysicalSample.id]
+  s3_path varchar [not null, unique]
+}
+
+Table Sequence {
+  id int [pk, increment]
+  gisaid_name varchar [unique]
+  gisaid_accession varchar [unique]
+
+  // TODO: Do we store the genome directly on this object?
+}
+
+// describes a single sequence that was uploaded to the system directly.
+Table UploadedSequence {
+  sequence_id int [not null, unique, ref: > Sequence.id]
+  physical_sample_id int [not null, ref: > PhysicalSample.id]
+}
+
+// describes a single sequence that was called from a consensus genome run
+Table ConsensusGenomeSequence {
+  sequence_id int [not null, unique, ref: > Sequence.id]
+  raw_genome_id int [not null, ref: > RawSequence.id]
+  
+  // TODO: metedata about the consensus genome run
+}
+
+///////////////////////////////////////////////////
+// Phylo Runs
+
+Table PhyloRun {
+  id int [pk, increment]
+}
```

Comment:
should we also have a field here that points to the location of the output JSONs?

Reply:
LOL right.  :)